synedrion/src/curve/arithmetic.rs

```diff
@@ -1,132 +1,169 @@
 use alloc::{format, string::String, vec, vec::Vec};
-use core::{
-    default::Default,
-    ops::{Add, Mul, Neg, Sub},
-};
+use core::ops::{Add, Mul, Neg, Rem, Sub};
+#[cfg(test)]
+use tiny_curve::TinyCurve32;
+use tiny_curve::TinyCurve64;
 
 use digest::XofReader;
-use k256::elliptic_curve::{
-    bigint::{U256, U512}, // Note that this type is different from typenum::U256
+use ecdsa::{SigningKey, VerifyingKey};
+use primeorder::elliptic_curve::{
+    bigint::{Concat, NonZero, Split, Zero},
     generic_array::{typenum::marker_traits::Unsigned, GenericArray},
+    group::{Curve as _, GroupEncoding},
     ops::Reduce,
     point::AffineCoordinates,
+    scalar::FromUintUnchecked,
     sec1::{EncodedPoint, FromEncodedPoint, ModulusSize, ToEncodedPoint},
     subtle::{Choice, ConditionallySelectable, CtOption},
-    Curve as _,
-    Field,
-    FieldBytesSize,
-    NonZeroScalar,
-    SecretKey,
-};
-use k256::{
-    ecdsa::{SigningKey, VerifyingKey},
-    elliptic_curve::group::ff::PrimeField,
-    Secp256k1,
+    Curve, CurveArithmetic, Field, FieldBytes, FieldBytesEncoding, FieldBytesSize, Group, NonZeroScalar, PrimeField,
+    ScalarPrimitive, SecretKey,
 };
+
 use rand_core::CryptoRngCore;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use serde_encoded_bytes::{Hex, SliceLike};
 use zeroize::Zeroize;
 
-use crate::tools::{
-    hashing::{Chain, HashableType},
-    Secret,
+use crate::{
+    tools::{
+        hashing::{Chain, HashableType},
+        Secret,
+    },
+    SchemeParams,
 };
 
-pub(crate) type Curve = Secp256k1;
-pub(crate) type BackendScalar = k256::Scalar;
-pub(crate) type BackendPoint = k256::ProjectivePoint;
-pub(crate) type CompressedPointSize = <FieldBytesSize<Secp256k1> as ModulusSize>::CompressedPointSize;
-
-pub(crate) const ORDER: U256 = Secp256k1::ORDER;
-
-impl HashableType for Curve {
+impl HashableType for k256::Secp256k1 {
```

Comment:
Can it be defined generically for any `Curve` implementor?

Reply:
I think problem here was that there's a blanket impl that conflicts? But maybe I'm wrong. I'll try!